tests/test_env_checker.py

```diff
@@ -112,3 +112,58 @@ def step(self, action):
     test_env = TestEnv()
     with pytest.raises(AssertionError, match=error_message):
         check_env(env=test_env)
+
+
+class StepCalledAfterEnvTerminatedException(Exception):
+    pass
+
+
+class LimitedStepsTestEnv(gym.Env):
+    metadata = {"render_modes": ["human"]}
+    render_mode = None
+
+    action_space = spaces.Discrete(n=2)
+    observation_space = spaces.Discrete(n=2)
+
+    def __init__(self, steps_before_termination: int = 1):
+        super().__init__()
+
+        assert steps_before_termination >= 1
+        self._steps_before_termination = steps_before_termination
+
+        self._steps_called = 0
+        self._terminated = False
+
+    def reset(self, *, seed: Optional[int] = None, options: Optional[Dict] = None) -> Tuple[int, Dict]:
+        super().reset(seed=seed)
+
+        self._steps_called = 0
+        self._terminated = False
+
+        return 0, {}
+
+    def step(self, action: np.ndarray) -> Tuple[int, float, bool, bool, Dict[str, Any]]:
+        self._steps_called += 1
+
+        if self._terminated:
+            raise StepCalledAfterEnvTerminatedException
+
+        observation = 0
+        reward = 0.0
+        self._terminated = self._steps_called >= self._steps_before_termination
+        truncated = False
+
+        return observation, reward, self._terminated, truncated, {}
+
+    def render(self, mode: str = "human") -> None:
```

Comment:
minor: mode argument has been removed in gymnasium

```suggestion
    def render(self) -> None:
```